From dcd72ff1425c2bb2089a69a6c75ef07fa9213b89 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Fri, 17 Jul 2026 11:13:20 +0200
Subject: [PATCH] netlink: bogus concatenated set ranges with netlink message
 overrun

JIRA: https://issues.redhat.com/browse/RHEL-190549
Upstream Status: nftables commit 2fbade3cd9900fe7f87ac660b6ac44544e238206

commit 2fbade3cd9900fe7f87ac660b6ac44544e238206
Author: Pablo Neira Ayuso <pablo@netfilter.org>
Date:   Thu Apr 17 21:40:23 2025 +0200

    netlink: bogus concatenated set ranges with netlink message overrun

    When building each component of the set element key, a late byteorder
    switch is performed to ensure that all components in the interval are
    represented in big endian, as required by the pipapo backend.

    In case that the set element does not fit into the netlink message, the
    byteorder switch happens twice, leading to inserting an element with a
    bogus component with large sets, so instead:

          "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890"

    listing reports:

      16777216 . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890"

    Note that 16777216 is 0x1000000, which should instead be 0x00000001 to
    represent "lo" as u32.

    Fix this by switching the value in a temporary variable and use it to
    set the set element key attribute in the netlink message.

    Later, revisit this to perform this byteorder switch from evaluation
    step.

    Add tests/shell unit to cover for this bug.

    Closes: https://bugzilla.netfilter.org/show_bug.cgi?id=1792
    Fixes: 8ac2f3b2fca3 ("src: Add support for concatenated set ranges")
    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 src/netlink.c                                 |  19 +-
 .../shell/testcases/sets/concat_nlmsg_overrun | 734 ++++++++++++++++++
 .../sets/dumps/concat_nlmsg_overrun.nft       |   7 +
 3 files changed, 755 insertions(+), 5 deletions(-)
 create mode 100755 tests/shell/testcases/sets/concat_nlmsg_overrun
 create mode 100644 tests/shell/testcases/sets/dumps/concat_nlmsg_overrun.nft

diff --git a/src/netlink.c b/src/netlink.c
index 8c191f1..1007571 100644
--- a/src/netlink.c
+++ b/src/netlink.c
@@ -263,6 +263,8 @@ static int __netlink_gen_concat_key(uint32_t flags, const struct expr *i,
 				    unsigned char *data)
 {
 	struct expr *expr;
+	mpz_t value;
+	int ret;
 
 	switch (i->etype) {
 	case EXPR_RANGE:
@@ -271,9 +273,11 @@ static int __netlink_gen_concat_key(uint32_t flags, const struct expr *i,
 		else
 			expr = i->left;
 
+		mpz_init_set(value, expr->value);
+
 		if (expr_basetype(expr)->type == TYPE_INTEGER &&
 		    expr->byteorder == BYTEORDER_HOST_ENDIAN)
-			byteorder_switch_expr_value(expr->value, expr);
+			byteorder_switch_expr_value(value, expr);
 
 		i = expr;
 		break;
@@ -294,22 +298,27 @@ static int __netlink_gen_concat_key(uint32_t flags, const struct expr *i,
 		}
 		return netlink_export_pad(data, i->prefix->value, i);
 	case EXPR_VALUE:
-		/* Switch byteorder only once for singleton values when the set
+		mpz_init_set(value, i->value);
+
+		/* Switch byteorder to big endian representation when the set
 		 * contains concatenation of intervals.
 		 */
-		if (!(flags & EXPR_F_INTERVAL))
+		if (!(flags & (EXPR_F_INTERVAL| EXPR_F_INTERVAL_END)))
 			break;
 
 		expr = (struct expr *)i;
 		if (expr_basetype(expr)->type == TYPE_INTEGER &&
 		    expr->byteorder == BYTEORDER_HOST_ENDIAN)
-			byteorder_switch_expr_value(expr->value, expr);
+			byteorder_switch_expr_value(value, expr);
 		break;
 	default:
 		BUG("invalid expression type '%s' in set", expr_ops(i)->name);
 	}
 
-	return netlink_export_pad(data, i->value, i);
+	ret = netlink_export_pad(data, value, i);
+	mpz_clear(value);
+
+	return ret;
 }
 
 static void netlink_gen_concat_key(const struct expr *expr,
diff --git a/tests/shell/testcases/sets/concat_nlmsg_overrun b/tests/shell/testcases/sets/concat_nlmsg_overrun
new file mode 100755
index 0000000..69cefe9
--- /dev/null
+++ b/tests/shell/testcases/sets/concat_nlmsg_overrun
@@ -0,0 +1,734 @@
+#!/bin/bash
+
+# NFT_TEST_REQUIRES(NFT_TEST_HAVE_pipapo)
+
+set -e
+
+RULESET='flush ruleset
+
+table ip filter {
+    set test_set {
+        type iface_index . ether_addr . ipv4_addr
+        flags interval
+            elements = {
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890",
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3,
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3,
+                    "lo" . 00:11:22:33:44:55 . 10.1.2.3,
+            }
+    }
+}'
+
+$NFT -f - <<< $RULESET
+
+exit 0
diff --git a/tests/shell/testcases/sets/dumps/concat_nlmsg_overrun.nft b/tests/shell/testcases/sets/dumps/concat_nlmsg_overrun.nft
new file mode 100644
index 0000000..01d76b9
--- /dev/null
+++ b/tests/shell/testcases/sets/dumps/concat_nlmsg_overrun.nft
@@ -0,0 +1,7 @@
+table ip filter {
+	set test_set {
+		type iface_index . ether_addr . ipv4_addr
+		flags interval
+		elements = { "lo" . 00:11:22:33:44:55 . 10.1.2.3 comment "123456789012345678901234567890" }
+	}
+}
